### PR TITLE
Add BatteryPool and PowerDistributingActor to the DataPipeline

### PR DESCRIPTION
**Pull request by sahas-subramanian-frequenz**

### CODE REVIEW

File: src/frequenz/sdk/microgrid/_data_pipeline.py [hunk 1]
```diff
@@ -123,6 +142,69 @@ def ev_charger_pool(
             )
         return self._ev_charger_pools[key]
 
+    def battery_pool(
+        self,
+        battery_ids: abc.Set[int] | None = None,
+    ) -> BatteryPool:
+        """Return the corresponding BatteryPool instance for the given ids.
+
+        If an BatteryPool instance for the given ids doesn't exist, a new one is
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/microgrid/_data_pipeline.py]:
```suggestion
        If a BatteryPool instance for the given ids doesn't exist, a new one is
```

File: src/frequenz/sdk/microgrid/_data_pipeline.py [hunk 2]
```diff
@@ -123,6 +142,69 @@ def ev_charger_pool(
...
+        created and returned.
+
+        Args:
+            battery_ids: Optional set of IDs of batteries to be managed by the
+                BatteryPool.
+
+        Returns:
+            An BatteryPool instance.
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/microgrid/_data_pipeline.py]:
```suggestion
            A BatteryPool instance.
```

File: src/frequenz/sdk/microgrid/_data_pipeline.py [hunk 3]
```diff
@@ -123,6 +142,69 @@ def ev_charger_pool(
...
+                batteries_status_receiver=self._battery_status_channel.new_receiver(),
+                min_update_interval=timedelta(
+                    seconds=self._resampler_config.resampling_period_s
+                ),
+                batteries_id=battery_ids,
+            )
+
+        return self._battery_pools[key]
+
+    def power_distributor_handle(self) -> Bidirectional.Handle[Request, Result]:
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/microgrid/_data_pipeline.py]: should this function  be named `power_distributing_handle` instead to be aligned with other function names or the actor name?

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have a couple of suggestions to check for, LGTM otherwise

File: src/frequenz/sdk/microgrid/_data_pipeline.py (hunk 1 above)
Comment (sahas-subramanian-frequenz) [reply src/frequenz/sdk/microgrid/_data_pipeline.py]: fixed

File: src/frequenz/sdk/microgrid/_data_pipeline.py (hunk 2 above)
Comment (sahas-subramanian-frequenz) [reply src/frequenz/sdk/microgrid/_data_pipeline.py]: fixed

File: src/frequenz/sdk/microgrid/_data_pipeline.py (hunk 3 above)
Comment (sahas-subramanian-frequenz) [reply src/frequenz/sdk/microgrid/_data_pipeline.py]: I've made this change for now.  This method is temporary anyway,  will go away as soon as BatteryPool has `charge`, `discharge` methods.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**